components/collective-page/hero/OverflowCollectiveName.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { truncate } from 'lodash';
+
+import LinkCollective from '../../LinkCollective';
+import StyledTooltip from '../../StyledTooltip';
+import { Span } from '../../Text';
+
+/**
+ * A tooltip that truncates a collective's name if it's longer than the
+ * provided cutOff.
+ */
+const OverflowCollectiveName = ({ collective, dataCy, cutOff = 30 }) => {
+  const collectiveName = collective.name;
+
+  // If cutOff is greater than 50 we'll truncate the string
+  // anyway we still have limited amount of space in the header
+  if (cutOff > 50) {
+    cutOff = 50;
+  }
+  // We also don't care about names shorter than 30 characters. Those
+  // should never get truncated.
+  if (collectiveName.length < 30) {
+    cutOff = 30;
+  }
+
+  return (
+    <LinkCollective collective={collective}>
```

Comment:
To make this generic, the <LinkCollective> should be done in the parent. This component should look more or less like that:

```jsx
if (value.length <= length) {
  return value;
} else {
  return (
    <React.Fragment>
      <StyledTooltip>...</StyledTooltip>
      {truncate(value, { length })}
    </React.Fragment>
}
```